src/tools/kata-ctl/src/ops/check_ops.rs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use crate::arch;
+use crate::check;
+use crate::ops::version;
+
+use crate::args::{
+    CheckArgument,
+    CheckSubCommand,
+    IptablesCommand,
+    MetricsCommand
+};
+
+use anyhow::Result;
+
+const NAME: &str = "kata-ctl";
+
+pub fn handle_check(checkcmd: CheckArgument) -> Result<()> {
+    let command = checkcmd.command;
+
+    match command {
+        CheckSubCommand::All => {
+            // run architecture-specific tests
+            let _all_checks = arch::check();
```

Comment:
I think we could just get these checks to return their error results as the stubs will return `Ok(())` so this could become:

```suggestion
            arch::check()?;
```

Reply:
@jodh-intel thank you, I was missing the return value of each function.